src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,191 @@
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car " + car + " .", e);
+        }
+        deleteDriversFromDb(car);
+        addDriversToDb(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car by id = " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long id) {
+        String query = "SELECT * FROM cars_drivers cd JOIN cars c ON cd.car_id = c.id "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE AND cd.driver_id = ?";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars by id = "
+                    + id + " .", e);
+        }
+        cars.forEach(car -> car.setDrivers(getDriversFromDb(car)));
+        return cars;
+    }
+
+    private void deleteDriversFromDb(Car car) {
+        String query = "DELETE FROM cars_drivers WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete drivers from DB by car "
+                    + car + " .", e);
+        }
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setModel(resultSet.getString("model"));
+        Long manufacturerId = resultSet.getObject("m.id", Long.class);
+        car.setId(resultSet.getObject("c.id", Long.class));
```

Comment:
```suggestion
        car.setId(resultSet.getObject("id", Long.class));
```
here better use alias, not c.id